map/mobile-map-search-and-route/src/main/java/com/esri/samples/mobile_map_search_and_route/MobileMapSearchAndRouteSample.java

```diff
@@ -213,77 +210,69 @@ protected void updateItem(ArcGISMap map, boolean bln) {
-                HashMap<String, Object> attributes = new HashMap<>();
-                attributes.put("title", address.split(",")[0]);
-                attributes.put("detail", address.substring(address.indexOf(", ") + 2));
-
-                // create a marker for the location
-                Graphic marker = new Graphic(geocode.getDisplayLocation(), attributes, pinSymbol);
-                locationsGraphicsOverlay.getGraphics().add(marker);
-
-                // display the callout at the location
-                Callout callout = mapView.getCallout();
-                callout.setTitle(marker.getAttributes().get("title").toString());
-                callout.setDetail(marker.getAttributes().get("detail").toString());
-                callout.setLeaderPosition(Callout.LeaderPosition.BOTTOM);
-                callout.showCalloutAt(location, new Point2D(0, -24), Duration.ZERO);
-
-                // if the map has transportation network datasets, solve for a route between the last two locations
-                if (!mapView.getMap().getTransportationNetworks().isEmpty() && locationsGraphicsOverlay.getGraphics().size() > 1) {
-                  // use the first transportation network dataset in the map to create a route task
-                  TransportationNetworkDataset networkDataset = mapView.getMap().getTransportationNetworks().get(0);
-                  RouteTask routeTask = new RouteTask(networkDataset);
-                  // create route parameters with the last two locations' stops
-                  List<Graphic> locationGraphics = locationsGraphicsOverlay.getGraphics();
-                  List<Stop> stops = locationGraphics.subList(Math.max(locationGraphics.size() - 2, 0), locationGraphics.size())
-                    .stream()
-                    .map(g -> new Stop((Point) g.getGeometry()))
-                    .collect(Collectors.toList());
-                  ListenableFuture<RouteParameters> defaultParameters = routeTask.createDefaultParametersAsync();
-                  defaultParameters.addDoneListener(() -> {
-                    try {
-                      RouteParameters routeParameters = defaultParameters.get();
-                      routeParameters.setStops(stops);
-                      // solve the route and display the result graphic
-                      ListenableFuture<RouteResult> routeResults = routeTask.solveRouteAsync(routeParameters);
-                      routeResults.addDoneListener(() -> {
-                        try {
-                          RouteResult result = routeResults.get();
-                          Route route = result.getRoutes().get(0);
+          mobileMapPackage.getLocatorTask().reverseGeocodeAsync(mapPoint, reverseGeocodeParameters).toCompletableFuture()
+            .whenComplete((geocodes, ex) -> {
+              if (ex == null) {
+                // show a pin graphic and a callout with the geocode's address
+                if (geocodes.size() > 0) {
```

Comment:
As commented for other sample - what if there is more than one result? How about commenting that we are only going to consider the first result in this sample?

Reply:
The implementation of `reverseGeocodeAsync` only returns one location:
<img width="1234" alt="image" src="https://github.com/Esri/arcgis-maps-sdk-java-samples/assets/67012037/f504453f-8e2d-4caf-aaf2-1cda0a2affec">

Could change in future, but I don't think that's worth commenting on. I'll add a comment that mentions that `geocodes` is a singleton list.